code/workspaces/infrastructure-api/src/controllers/projectsController.js

```diff
@@ -0,0 +1,96 @@
+import { check, matchedData } from 'express-validator';
+import { service } from 'common';
+import projectsRepository from '../dataaccess/projectsRepository';
+
+async function listProjects(request, response, next) {
+  try {
+    const projects = await projectsRepository.getAll() || [];
+    response.send(projects);
+  } catch (error) {
+    next(new Error(`Error listing projects: ${error.message}`));
+  }
+}
+
+async function getProjectByKey(request, response, next) {
+  const { key } = matchedData(request);
+
+  try {
+    const project = await projectsRepository.getByKey(key);
+    if (project) response.send(project);
+    else response.status(404).send();
+  } catch (error) {
+    next(new Error(`Error getting project by key: ${error.message}`));
+  }
+}
+
+async function createProject(request, response, next) {
+  const document = request.body;
+
+  if (await projectsRepository.exists(document)) {
+    response.status(400).send({
+      errors: [{ msg: `Entry with projectKey '${document.projectKey}' already exists.` }],
+    });
+  } else {
+    try {
+      const project = await projectsRepository.create(document);
+      response.status(201).send(project);
+    } catch (error) {
+      next(new Error(`Error creating project: ${error.message}`));
+    }
+  }
+}
+
+async function createOrUpdateProject(request, response, next) {
+  const { key } = matchedData(request);
+  const document = request.body;
+
+  if (key !== document.projectKey) {
+    const msg = "'key' parameter in URL and 'projectKey' field in request body must "
+      + `match. Got key='${key}' and projectKey='${document.projectKey}'.`;
+    response.status(400).send({ errors: [{ msg }] });
+  } else {
+    try {
+      // Doesn't return object if it is created hence gets by key if there is no return
```

Comment:
I think there is flag you can set to make it return the created object. In the repository code set `{ new: true }` on the options.
https://davidburgos.blog/return-updated-document-mongoose/

Reply:
Perfect, thank you for pointing that out!